### PR TITLE
Feature/uche/hook edit item backend

### PR DESCRIPTION
**Pull request by enyeInc**
*warning* - for now do not create items for you are not logged in

- User are not able to create new items
- User are now able to view items that they created

### CODE REVIEW

File: routes/product/helpers.js [hunk 1]
```diff
@@ -89,15 +108,20 @@ export default {
...
+                await Product.updateOne(filter, product);
+
+                data = product;
+                message = PRODUCT_UPDATED;
             } else {
-                // Redundant code to catch if user insert invalid '_id'
+                // Redundancy: deletes possible invalid/valid '_id'
                 delete product._id;
 
                 data = await Product.create(product);
```
Comment (EngrEric) [inline routes/product/helpers.js]: how is this possible, passing the new `productData` as param to the instance of the Product model. I need clarifications here

File: routes/product/helpers.js [hunk 2]
```diff
@@ -89,15 +108,20 @@ export default {
                         { producerId: { $eq: producerId } },
                     ],
                 };
```
Comment (EngrEric) [inline routes/product/helpers.js]: what is this filter doing and what is it filtering

**Comment by EngrEric** [comment]
Looks very good to me

File: routes/product/helpers.js (hunk 2 above)
Comment (enyeInc) [reply routes/product/helpers.js]: we are filter products where ` _id is equal to _id and producerId is not null and producerId is equal to producerId`. It is finding exactly the product that we need to updated

File: routes/product/helpers.js (hunk 1 above)
Comment (enyeInc) [reply routes/product/helpers.js]: So there are many ways (i think about 3 or so) to create a new item using mongoose. This is just want of the ways

**Pull request merged**